src/comodo/drakeSimulator/utils.py

```diff
@@ -0,0 +1,209 @@
+        if not os.path.exists(self.urdf_out_path):
+            os.makedirs(self.urdf_out_path)
+        filename = self.urdf_out_path + "/" + urdf_name + ".urdf"
+        with open(filename, "w") as f:
+            print(self.odio_robot, file=f)
+
+        logging.info("Saved the urdf to {}".format(filename))
+
+    def get_urdf_string(self):
+        """Returns the urdf string"""
+        return str(self.odio_robot)
+
+    def get_sim_joint_order(self, plant, robot_model_sim):
+        sim_joint_order = []
+        for ii in plant.GetJointIndices(robot_model_sim):
+            jj = plant.get_joint(ii)
+            if jj.type_name() == "revolute":
+                sim_joint_order.append(jj.name())
+        return sim_joint_order
+
+    # add ground with friction
+    def add_ground_with_friction(self, plant):
+        surface_friction_ground = CoulombFriction(
+            static_friction=1.0, dynamic_friction=1.0
+        )
+        proximity_properties_ground = ProximityProperties()
+        AddContactMaterial(
+            1e4, 1e7, surface_friction_ground, proximity_properties_ground
+        )
+        AddRigidHydroelasticProperties(0.01, proximity_properties_ground)
+
+        plant.RegisterCollisionGeometry(
+            plant.world_body(),
+            RigidTransform(),
+            HalfSpace(),
+            "ground_collision",
+            proximity_properties_ground,
+        )
+        plant.RegisterVisualGeometry(
+            plant.world_body(),
+            RigidTransform(),
+            HalfSpace(),
+            "ground_visual",
+            np.array([0.5, 0.5, 0.5, 0.0]),
+        )
+
+    def add_soft_feet_collisions(self, plant, xMinMax, yMinMax):
+        surface_friction_feet = CoulombFriction(
+            static_friction=1.0, dynamic_friction=1.0
+        )
+        proximity_properties_feet = ProximityProperties()
+        AddContactMaterial(1e4, 1e7, surface_friction_feet, proximity_properties_feet)
+        AddCompliantHydroelasticProperties(0.01, 1e8, proximity_properties_feet)
+        for ii in ["l_foot_front", "l_foot_rear", "r_foot_front", "r_foot_rear"]:
+            # for ii in ["r_foot", "l_foot"]:
+            # for collision
+            plant.RegisterCollisionGeometry(
+                plant.GetBodyByName(ii),
+                RigidTransform(np.array([0, 0, 0])),
+                # RigidTransform(),
+                BoxDrake((xMinMax[1] - xMinMax[0]) / 2, yMinMax[1] - yMinMax[0], 2e-3),
+                ii + "_collision",
+                proximity_properties_feet,
+            )
+
+            # for visual
+            plant.RegisterVisualGeometry(
+                plant.GetBodyByName(ii),
+                RigidTransform(np.array([0, 0, 0])),
+                # RigidTransform(),
+                BoxDrake((xMinMax[1] - xMinMax[0]) / 2, yMinMax[1] - yMinMax[0], 2e-3),
+                ii + "_collision",
+                np.array([1.0, 1.0, 1.0, 1]),
```

Comment:
What is this value ?

Reply:
done in b53d0091f81dc73c27952c6bdcb393d72d846767 ( they are RGBA values for the feet )